semantics/src/namespace/scopes.rs

```diff
@@ -287,10 +325,76 @@ mod tests {
         let contract_scope = ContractScope::new(module_scope);
         let block_scope_1 =
             BlockScope::from_contract_scope(Span::new(0, 0), contract_scope.clone());
-        let block_scope_2 = BlockScope::from_block_scope(Span::new(0, 0), block_scope_1.clone());
+        let block_scope_2 = BlockScope::from_block_scope(
+            Span::new(0, 0),
+            BlockScopeType::IfElse,
+            block_scope_1.clone(),
```

Comment:
I had been using `Rc::clone(&scope)`, since that's what's used in the API docs for Rc: https://doc.rust-lang.org/std/rc/struct.Rc.html.

Both of these do the same thing, of course, and I don't have a strong opinion about which method we use to call `clone`, but I think we should be consistent.

Reply:
Thanks for bringing this up. I wasn't aware that there was a good reason for the the more verbose syntax. I looked it up and `RC::clone(&ptr)` is indeed the preferred syntax. The reason for that is to make it clear what the `clone` is about since `clone` is a very vaguely defined action in Rust. In this case it is just increasing the reference counter whereas e.g. cloning a `Vec<T>` can cause quite a bit of memory getting copied.

References:

https://github.com/nical/rfcs/blob/rc-newref-clone/text/0000-rc-newref-clone.md
https://github.com/rust-lang/rust/pull/42137#issuecomment-304908345

I'm going to merge this as is and send a follow up PR to rewrite all occurrences with the idiomatic syntax.

Reply:
Updated all usage in this PR https://github.com/ethereum/fe/pull/126